temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java

```diff
@@ -139,34 +168,40 @@ public boolean isTerminated() {
 
   @Override
   public CompletableFuture<Void> shutdown(ShutdownManager shutdownManager, boolean interruptTasks) {
-    return poller.shutdown(shutdownManager, interruptTasks);
+    return stickyPoller != null
+        ? CompletableFuture.allOf(
+            poller.shutdown(shutdownManager, interruptTasks),
+            stickyPoller.shutdown(shutdownManager, interruptTasks))
+        : poller.shutdown(shutdownManager, interruptTasks);
   }
 
   @Override
   public void awaitTermination(long timeout, TimeUnit unit) {
-    poller.awaitTermination(timeout, unit);
+    long timeoutMillis = ShutdownManager.awaitTermination(poller, unit.toMillis(timeout));
+    ShutdownManager.awaitTermination(stickyPoller, timeoutMillis);
```

Comment:
In other methods you are concerned `stickyPoller` may be `null`, no similar concerns here? Is `awaitTermination` null-safe?

Reply:
It's null safe, but a good note. I will add a check just for uniformity.